ehcache-impl/src/test/java/org/ehcache/impl/internal/store/heap/bytesized/OnHeapStoreByteSizeTest.java

```diff
@@ -0,0 +1,106 @@
+package org.ehcache.impl.internal.store.heap.bytesized;
+
+import org.ehcache.Cache;
+import org.ehcache.CacheManager;
+import org.ehcache.config.CacheConfiguration;
+import org.ehcache.config.builders.CacheConfigurationBuilder;
+import org.ehcache.config.builders.CacheManagerBuilder;
+import org.ehcache.config.builders.ExpiryPolicyBuilder;
+import org.ehcache.config.builders.ResourcePoolsBuilder;
+import org.ehcache.config.units.MemoryUnit;
+import org.ehcache.core.internal.statistics.DefaultStatisticsService;
+import org.ehcache.core.internal.statistics.DefaultTierStatistics;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.time.Duration;
+
+import static java.lang.Integer.parseInt;
+import static java.lang.System.getProperty;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assume.assumeThat;
+
+public class OnHeapStoreByteSizeTest {
+
+  @BeforeClass
+  public static void preconditions() {
+    assumeThat(parseInt(getProperty("java.specification.version").split("\\.")[0]), is(lessThan(16)));
+  }
+
+  @Test
+  public void testByteSizePostClearAndRemoveUsingDefaultStatisticsService() {
+
+    DefaultStatisticsService statisticsService = new DefaultStatisticsService();
+    CacheManager cacheManager = CacheManagerBuilder.newCacheManagerBuilder()
```

Comment:
Would be better code-style wise to use try-with-resources blocks for both the CacheManager and the Cache here, e.g.:
```java
try (CacheManager cacheManager = newCacheManagerBuilder().using(statisticsService).build(true)) {
  ...
}
```